### PR TITLE
Force utf-8 encoding before scrubbing

### PR DESCRIPTION
**Pull request by ckumar1**
Fixes the encoding issues introduced by #23 

/cc @fneves @ggrossman @grosser

### CODE REVIEW

**Comment by grosser** [comment]
test ? :)

:+1:

File: test/credit_card_sanitizer_test.rb [hunk 1]
```diff
@@ -66,6 +66,11 @@ class CreditCardSanitizerTest < MiniTest::Test
         end
       end
 
+      it "doesn't fail if text is not utf-8 encoded" do
+        ascii_text = '41111111111111112'.force_encoding(Encoding::ASCII)
```
Comment (grosser) [inline test/credit_card_sanitizer_test.rb]: this test also passed without the new code, the problem we are trying to solve is that a with invalid utf8 that is encoded in another encoding was not stripped correctly

```
      it "doesn't fail if the text contains invalid utf-8 characters but is not in utf-8" do
        if ''.respond_to?(:encoding)
          invalid_characters = "你好 4111 1111 1111 1111 \255there".force_encoding(Encoding::BINARY)
          assert_equal "你好 4111 11▇▇ ▇▇▇▇ 1111 \ufffdthere", @sanitizer.sanitize!(invalid_characters)
        end
      end
```

Comment (ckumar1) [inline test/credit_card_sanitizer_test.rb]: I was getting the following error without the new code in the test:

```
  1) Error:
CreditCardSanitizer::#sanitize!#test_0010_doesn't fail if text is not utf-8 encoded:
Encoding::CompatibilityError: incompatible character encodings: US-ASCII and UTF-8
    /Users/ckumar/Code/zendesk/credit_card_sanitizer/lib/credit_card_sanitizer.rb:64:in `scrub!'
    /Users/ckumar/Code/zendesk/credit_card_sanitizer/lib/credit_card_sanitizer.rb:64:in `sanitize!'
    ./test/credit_card_sanitizer_test.rb:71:in `block (3 levels) in <class:CreditCardSanitizerTest>'
```
But I can update the test to check for stripping out the invalid characters

Comment (grosser) [inline test/credit_card_sanitizer_test.rb]: Sweet never mind then :)

**Pull request merged**